compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -741,25 +729,374 @@ fn encode_ty<'tcx>(
     typeid
 }
 
+/// Returns the list of arguments if principal is an Fn trait.
```

Comment:
In general, this seems like a lot of helper functions that do a lot of things. I think they can all be inlined into each other and simplified quite a lot.

Reply:
Done.